vmm/src/cpu.rs

```diff
@@ -876,40 +875,71 @@ impl CpuManager {
             creation_ts,
         )?;
 
-        let reset_evt = self.reset_evt.try_clone().unwrap();
-        let vcpu_kill_signalled = self.vcpus_kill_signalled.clone();
-        let vcpu_pause_signalled = self.vcpus_pause_signalled.clone();
-
-        let vcpu_kill = self.vcpu_states[usize::from(cpu_id)].kill.clone();
-        let vcpu_run_interrupted = self.vcpu_states[usize::from(cpu_id)]
-            .vcpu_run_interrupted
-            .clone();
-
         if let Some(snapshot) = snapshot {
-            let mut cpuid = self.cpuid.clone();
-            CpuidPatch::set_cpuid_reg(&mut cpuid, 0xb, None, CpuidReg::EDX, u32::from(cpu_id));
-
-            vcpu.lock()
-                .unwrap()
-                .fd
-                .set_cpuid2(&cpuid)
-                .map_err(Error::SetSupportedCpusFailed)?;
-
+            #[cfg(target_arch = "x86_64")]
+            {
+                let mut cpuid = self.cpuid.clone();
+                CpuidPatch::set_cpuid_reg(&mut cpuid, 0xb, None, CpuidReg::EDX, u32::from(cpu_id));
+
+                vcpu.lock()
+                    .unwrap()
+                    .fd
+                    .set_cpuid2(&cpuid)
+                    .map_err(Error::SetSupportedCpusFailed)?;
+            }
             vcpu.lock()
                 .unwrap()
                 .restore(snapshot)
                 .expect("Failed to restore vCPU");
         } else {
             let vm_memory = self.vm_memory.clone();
 
+            #[cfg(target_arch = "x86_64")]
             vcpu.lock()
                 .unwrap()
                 .configure(entry_point, &vm_memory, self.cpuid.clone())
                 .expect("Failed to configure vCPU");
+
+            #[cfg(target_arch = "aarch64")]
+            vcpu.lock()
+                .unwrap()
+                .configure(entry_point, &vm_memory)
+                .expect("Failed to configure vCPU");
         }
 
         let vcpu_clone = Arc::clone(&vcpu);
-        self.vcpus.push(vcpu_clone);
+
+        Ok(vcpu_clone)
+    }
+
+    fn create_vcpus(&mut self, desired_vcpus: u8, entry_point: Option<EntryPoint>) -> Result<()> {
+        if desired_vcpus > self.max_vcpus {
+            return Err(Error::DesiredVCPUCountExceedsMax);
+        }
+
+        for cpu_id in self.present_vcpus()..desired_vcpus {
+            let vcpu = self.create_vcpu(cpu_id, entry_point, None).unwrap();
```

Comment:
Don't use `.unwrap()` as that will panic on the error. Propagate up with `?` There are a few other places where you do this too.

Reply:
Thanks. 
Updated.